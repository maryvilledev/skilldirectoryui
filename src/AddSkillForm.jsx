import React, { PropTypes } from 'react';
import { Col, ControlLabel, Button,
   Form, FormControl, FormGroup } from 'react-bootstrap'

const skillTypes = [
  { value: 'compiled', label: 'Compiled' },
  { value: 'scripted', label: 'Scripted' },
  { value: 'database', label: 'Database' },
  { value: 'orchestration', label: 'Orchestration' }
];

const options = skillTypes.map((type, index) => {
  return (
    <option key={index} value={type.value}>
      {type.label}
    </option>
  );
});

class AddSkillForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      skill_name: "",
      skill_type: skillTypes[0].value
    };
    this.onChange = this.onChange.bind(this);
    this.onNameChange = this.onNameChange.bind(this);
    this.onTypeChange = this.onTypeChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(key) {
    return (value) => {
      this.setState({
        [key]: value
      });
    }
  }

  onNameChange(ev) {
    this.onChange('skill_name')(ev.target.value);
  }

  onTypeChange(ev) {
    console.log('addskillform typechange ev:')
    console.log(ev);
    this.onChange('skill_type')(ev.target.value);
  }

  onSubmit(ev) {
    ev.preventDefault();
    this.props.onSubmit(
      this.state.skill_name,
      this.state.skill_type
    );
  }

  render() {
    return (
      <div>
        <h1>Add Skill</h1>
        <Form horizontal onSubmit={this.onSubmit}>
          <FormGroup controlId='skillName'>
            <Col componentClass={ControlLabel} sm={2}>
              Name:
            </Col>
            <Col sm={10}>
              <FormControl name='skillName' onChange={this.onNameChange}/>
            </Col>
          </FormGroup>
          <FormGroup controlId='skillType'>
            <Col componentClass={ControlLabel} sm={2}>
              Type:
            </Col>
            <Col sm={10}>
              <FormControl componentClass="select" onChange={this.onTypeChange}>
                {options}
              </FormControl>
            </Col>
          </FormGroup>
          <FormGroup>
            <Col smOffset={2} sm={1}>
              <Button type="submit" bsStyle="primary">
                Submit
              </Button>
            </Col>
            <Col smOffset={1} sm={2}>
              <Button onClick={this.props.onCancel} bsStyle="info">
                Cancel
              </Button>
            </Col>
          </FormGroup>
        </Form>
      </div>
    );
  }
}

AddSkillForm.propTypes = {
  onCancel: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default AddSkillForm;
