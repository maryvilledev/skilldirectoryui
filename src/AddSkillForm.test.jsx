import React from 'react';
import ReactDOM from 'react-dom';
import AddSkillForm from './AddSkillForm.jsx';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';

describe('<AddSkillForm />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddSkillForm />, div);
  });

  it('matches the stored snapshot', () => {
    const wrapper = shallow(<AddSkillForm />);
    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('changes state when input to "Name:" field changes', () => {
    const wrapper = mount(<AddSkillForm />);
    // process.stdout.write(wrapper.debug());
    // process.stdout.write(JSON.stringify(wrapper.state()));
    expect(wrapper.state().skill_name).toBe("");
    const event = {target: {value: "New Value"}};
    wrapper.find({ name: "skillName" }).simulate("change", event);
    expect(wrapper.state().skill_name).toBe("New Value");
    // process.stdout.write(JSON.stringify(wrapper.state()));
  });

// TODO: Find different way to test this... The below test fails because
// the "Select" element we are using from 'react-select' doesn't respond
// as expected to the simulated "change" event.
  // it('changes state when selected option in "Skill Type" field changes', () => {
  //   const wrapper = mount(<AddSkillForm />);
  //   expect(wrapper.state().skill_type).toBe("");
  //   const event = {target: {value: "New Value"}};
  //   wrapper.find("Select").simulate("change", event);
  //   expect(wrapper.state().skill_type).toBe("New Value");
  // });
});
