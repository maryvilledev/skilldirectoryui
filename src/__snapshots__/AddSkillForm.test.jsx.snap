exports[`<AddSkillForm /> matches the stored snapshot 1`] = `
<div>
  <h1>
    Add Skill
  </h1>
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={true}
    inline={false}
    onSubmit={[Function]}>
    <FormGroup
      bsClass="form-group"
      controlId="skillName">
      <Col
        bsClass="col"
        componentClass={[Function]}
        sm={2}>
        Name:
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        sm={10}>
        <FormControl
          bsClass="form-control"
          componentClass="input"
          name="skillName"
          onChange={[Function]} />
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      controlId="skillType">
      <Col
        bsClass="col"
        componentClass={[Function]}
        sm={2}>
        Type:
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        sm={10}>
        <FormControl
          bsClass="form-control"
          componentClass="select"
          onChange={[Function]}>
          <option
            value="compiled">
            Programming Language (Compiled)
          </option>
          <option
            value="scripted">
            Programming Language (Scripted)
          </option>
          <option
            value="database">
            Database Language
          </option>
          <option
            value="orchestration">
            DevOps Tool
          </option>
        </FormControl>
      </Col>
    </FormGroup>
    <FormGroup
      bsClass="form-group">
      <Col
        bsClass="col"
        componentClass="div"
        sm={1}
        smOffset={2}>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          type="submit">
          Submit
        </Button>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        sm={2}
        smOffset={1}>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="info"
          disabled={false}
          onClick={undefined}>
          Cancel
        </Button>
      </Col>
    </FormGroup>
  </Form>
</div>
`;
